README.md

```diff
@@ -207,23 +220,273 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+        if (any(nullIndices)) {
+            class$methods <- culledMethods[-which(nullIndices)]
+        }
+    }
+    class
+}
+```
+
+Then call `generateRdFiles` and `generateRWrappers` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseclient",
+                module = "synapseclient.entity",
+                modifyClasses = selectClasses)
+```
+```r
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.entity",
+                  setGenericCallback = callback,
+                  modifyClasses = selectClasses)
+```
+
+### Expose functions within a singleton object
+
+In rare cases, we want to expose Python functions under `synapseclient.client.Synapse` without exposing the `Synapse` object itself.
+
+In a Python session, we would do the following:
+```
+  syn = synapseclient.Synapse()
+  syn.login()
+  syn.get()
+```
+
+And in R, users would not need to know about the `Synapse` object:
+```
+  synLogin()
+  synGet()
+```
+
+Note that:
+* all function calls in R access the same underlying Python object `Synapse`
+* the functions name in R has a `syn` prefix
+
+To do this, one would specifying `pyObjectName` and `functionPrefix` params as following:
+
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseclient",
+                module = "synapseclient.client.Synapse",
+                pyObjectName = "syn",
+                functionPrefix = "syn")
+```
+```r
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.client.Synapse",
+                  setGenericCallback = callback,
+                  pyObjectName = "syn",
+                  functionPrefix = "syn")
+```
+
+Then in the same `.onLoad`, create the Python object:
+```r
+pyImport("synapseclient")
+pyExec("syn = synapseclient.Synapse()")
+```
+
+### Overwrite the returned object in R
+
+For many reasons, a returned object from an R wrapper function does not have the R-friendly bahaviors. As an author of an R package, one may want to modify the returned object. Here is a simple example using `transformReturnObject` parameter:
+```r
+objectDefinitionHelper <- function(object) {
+  # change returned object name from "GeneratorWrapper.<some function name>" to "GeneratorWrapper"
+  if (grepl("^GeneratorWrapper", class(object)[1])) {
+    class(object)[1] <- "GeneratorWrapper"
+  }
+  object
+}
+
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.client.Synapse",
+                  setGenericCallback = callback,
+                  functionPrefix = "syn",
+                  pyObjectName = "syn",
+                  transformReturnObject = objectDefinitionHelper)
+```
+
+### Notes on `generateRdFiles`
+
+This helper function can be used to convert Python [Sphinx](http://www.sphinx-doc.org/en/master/) docs into Rd files. It only supports a subset of Sphinx tags, including:
+- title
+- description
+- usage
+- arguments
+- value
+- examples
```

Comment:
You've listed the .Rd sections, not the sphinx tags, which include `:param:, :type:, :return:, :returns:`.